File: cpp_version/Trawa.cpp
#include "Trawa.h"
#include "Swiat.h"

Trawa::Trawa(int x, int y, Swiat& swiat)
    : Roslina(0, x, y, swiat) {
}

char Trawa::rysowanie() const { return 'T'; }

std::string Trawa::getNazwa() const { return "Trawa"; }

std::shared_ptr<Organizm> Trawa::stworzPotomka(int x, int y) {
    return std::make_shared<Trawa>(x, y, swiat);
}

void Trawa::zasiej(int nowyX, int nowyY) {
    swiat.dodajOrganizm(stworzPotomka(nowyX, nowyY));
}

File: java_version/Trawa.java
public class Trawa extends Roslina {
    public Trawa(int x, int y, Swiat swiat) {
        super(0, x, y, swiat);
    }

    @Override
    public char rysowanie() {
        return 'T';
    }

    @Override
    public String getNazwa() {
        return "Trawa";
    }

    @Override
    public Organizm stworzPotomka(int x, int y) {
        return new Trawa(x, y, swiat);
    }

    @Override
    protected void zasiej(int nowyX, int nowyY) {
        swiat.dodajOrganizm(stworzPotomka(nowyX, nowyY));
    }
}
